src/main/java/codeu/controller/ProfileServlet.java

```diff
@@ -5,14 +5,113 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import codeu.model.data.User;
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.util.UUID;
+import java.util.List;
+import codeu.model.store.basic.UserStore;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
+import codeu.model.store.basic.UserStore;
+import codeu.model.data.User;
+import codeu.model.data.Message;
 
-
+import org.mindrot.jbcrypt.BCrypt;
 /** Servlet class responsible for the Profile page. */
 public class ProfileServlet extends HttpServlet {
+	
+	private UserStore userStore;
+	
+	private ConversationStore conversationStore;
+	
+	private MessageStore messageStore;
+	
+	@Override
+	  public void init() throws ServletException {
+	    super.init();
+	    setUserStore(UserStore.getInstance());
+	    setConversationStore(ConversationStore.getInstance());
+	    setMessageStore(MessageStore.getInstance());
+	  }
+
+	  /**
+	   * Sets the UserStore used by this servlet. This function provides a common setup method for use
+	   * by the test framework or the servlet's init() function.
+	   */
+	  void setUserStore(UserStore userStore) {
+	    this.userStore = userStore;
+	  }
+	  
+	  void setConversationStore(ConversationStore conversationStore) {
+		  this.conversationStore = conversationStore;
+	  }
+	  
+	  void setMessageStore(MessageStore messageStore) {
+		  this.messageStore = messageStore;
+	  }
+	  
+	  
+	  
+	  
 	@Override
-	public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
-		request.getRequestDispatcher("/WEB-INF/view/profile.jsp").forward(request, response);
+	public void doGet(HttpServletRequest request, HttpServletResponse response) 
+			throws IOException, ServletException {
+				String requestUrl = request.getRequestURI();
+				String userProfile = requestUrl.substring("/users/".length());
+				
+				User user = userStore.getUser(userProfile);
+				if (user == null) {
+					response.sendRedirect("/login");
+					return;
+				}
+		
+			request.setAttribute("user", user);
+			request.setAttribute("userProfile", userProfile);
+				
+			UUID writer = user.getId();
+			MessageStore message = MessageStore.getInstance();
+			List<Message> messagesSent = message.getMessageOfUser(writer); //get the users messages
```

Comment:
Comments should go above the line of code. Put a space between // and the first character and a . at the end!

Reply:
this still applies!